pinot-server/src/main/java/org/apache/pinot/server/api/resources/HealthCheckResource.java

```diff
@@ -55,7 +55,36 @@ public class HealthCheckResource {
       @ApiResponse(code = 503, message = "Server is not healthy")
   })
   public String checkHealth() {
```

Comment:
Consider adding a deprecate annotation with some comment

Reply:
per discussion in https://github.com/apache/pinot/issues/9023 I think we should skill keep this. 
I will merge for now and if needed we can come up with a plan for deprecate.